unitstatushandler/unitstatushandler.go

```diff
@@ -185,6 +194,31 @@ func (instance *Instance) Close() (err error) {
 	return aoserrors.Wrap(err)
 }
 
+// SendInitialUnitStaus send initial unit status.
+func (instance *Instance) SendInitialUnitStaus() error {
+	if err := instance.initCurrentStatus(); err != nil {
+		return aoserrors.Wrap(err)
+	}
+
+	instance.sendCurrentStatus()
+
+	return nil
+}
+
+// ProcessRunStatus process current run instances status.
+func (instance *Instance) ProcessRunStatus(status RunInstancesStatus) {
+	instance.unitSubjects = status.UnitSubjects
+
+	instance.softwareManager.processRunStatus(status)
+	instance.sendCurrentStatus()
+}
+
+// ProcessUpdateInstanceStatus process update instances status.
+func (instance *Instance) ProcessUpdateInstanceStatus(status []cloudprotocol.InstanceStatus) {
```

Comment:
`ProcessUpdateInstancesStatus`. This function should pass instance statuses to the function `updateInstanceStatus` which should be similar to `updateServiceStatus`.